### PR TITLE
[WR-1605] add ssotFormat function to handle NULL/0, use ssotFormat function for…

### PR DESCRIPTION
**Pull request by rjkuyvenhoven**
… Industry Employment by Region table

I found an table that need null checking.  I created an ssotFormat function to do the checking and formatting.  I created a new function that calls ssotFormatNumber function rather than adding the null checking to ssotFormatNumber to avoid the possibility of introduction new bugs into the system.

### CODE REVIEW

File: src/web/modules/custom/workbc_extra_fields/src/Plugin/ExtraField/Display/CareerProfile/CareerProfileJobOpeningsByRegion.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function viewElements(ContentEntityInterface $entity) {
       $regions[] = $region;
       $region = array();
       $region['name'] = t(REGION_VANCOUVER_ISLAND_COAST);
-      $rvalue = $entity->ssot_data['career_regional']['vancouver_island_coast_expected_number_of_job_openings_10y'];
+      $value = $entity->ssot_data['career_regional']['vancouver_island_coast_expected_number_of_job_openings_10y'];
       $region['openings'] = ($value===0||$value) ? ssotFormatNumber(floatval($value),0) : WORKBC_EXTRA_FIELDS_NOT_AVAILABLE;
```
Comment (infojunkie) [inline src/web/modules/custom/workbc_extra_fields/src/Plugin/ExtraField/Display/CareerProfile/CareerProfileJobOpeningsByRegion.php:93]: is this also a candidate to reuse `ssotFormat()` ?

File: src/web/modules/custom/workbc_extra_fields/src/Plugin/ExtraField/Display/CareerProfile/CareerProfileJobOpeningsByRegion.php [hunk 2]
```diff
@@ -89,7 +89,7 @@ public function viewElements(ContentEntityInterface $entity) {
...
       $value = $entity->ssot_data['career_regional']['vancouver_island_coast_annual_employment_growth_10y_pct'];
       $region['growth'] = ($value===0||$value) ? ssotFormatNumber($value,1) . "%" : WORKBC_EXTRA_FIELDS_NOT_AVAILABLE;
```
Comment (infojunkie) [inline src/web/modules/custom/workbc_extra_fields/src/Plugin/ExtraField/Display/CareerProfile/CareerProfileJobOpeningsByRegion.php:95]: is this also a candidate to reuse `ssotFormat()` ?

File: src/web/modules/custom/workbc_extra_fields/workbc_extra_fields.module [hunk 3]
```diff
@@ -373,6 +373,17 @@ function ssotIndustryName($machine_name) {
   return isset($industries[$machine_name]) ? $industries[$machine_name] : "";
 }
 
+function ssotFormat($number, $decimals = 0, $suffix = "") {
```
Comment (infojunkie) [inline src/web/modules/custom/workbc_extra_fields/workbc_extra_fields.module:376]: I suggest to call this `ssotFormatNA`

Comment (rjkuyvenhoven) [reply src/web/modules/custom/workbc_extra_fields/workbc_extra_fields.module:376]: I like that suggestion

**Comment by rjkuyvenhoven** [comment]
CareerJobOpeningsByRegion.php updated to use ssotFormatNA()
CareerJobOpeningsByRegionNA.php updated to remove floatval() so function will accurately return YES|NO if data contains null or not.

**Review by infojunkie — APPROVED** [review]
Lgtm, thanks for making this refactoring

**Pull request merged**